react/AddressForm.tsx

```diff
@@ -1,46 +1,162 @@
-import { Input } from 'vtex.styleguide'
+import React, { useState } from 'react'
+import { Input, Dropdown, ButtonPlain } from 'vtex.styleguide'
 import { useAddressContext } from 'vtex.address-context/AddressContext'
-import { LineFragment } from './typings/countryRulesTypes'
+import {
+  LineFragment,
+  Fields,
+  OptionsField,
+  Display,
+} from './typings/countryRulesTypes.d'
+import rules from './countries/rules'
+import PlaceDetails from './PlaceDetails'
+import { FormattedMessage, defineMessages } from 'react-intl'
+import NumberOption from './components/NumberOption'
 
-interface Props {
-  address: Address
-  summary: LineFragment[][]
-}
+defineMessages({
+  country: {
+    defaultMessage: '',
+    id: 'place-components.label.country',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  neighborhood: {
+    defaultMessage: '',
+    id: 'place-components.label.neighborhood',
+  },
+  ['street-road-avenue']: {
+    defaultMessage: '',
+    id: 'place-components.label.street-road-avenue',
+  },
+  ['number-option']: {
+    defaultMessage: '',
+    id: 'place-components.label.number-option',
+  },
+  complement: {
+    defaultMessage: '',
+    id: 'place-components.label.complement',
+  },
+  stateAbbreviation: {
+    defaultMessage: '',
+    id: 'place-components.label.stateAbbreviation',
+  },
+})
+
+const AddressForm: StorefrontFunctionComponent<{}> = () => {
+  const { address, setAddress } = useAddressContext()
+  const [editing, setEditing] = useState<boolean>(false)
+  const { fields, display } = rules[address.country]
+  const summary = display['extended'] as LineFragment[][]
+
+  const getSummaryFields = (summary: LineFragment[][]) => {
+    let summaryFields = new Set()
+    summary.forEach((line: LineFragment[]) => {
+      line.forEach((fragment: LineFragment) => {
+        summaryFields.add(fragment.name)
+      })
+    })
+    return summaryFields
+  }
+
+  const [ignoredFields, setIgnoredFields] = useState(
+    getSummaryFields(display['compact'])
+  )
+
+  const numberHasWithoutOption = (label: string | null) => {
+    return (
+      label && label.length >= 7 && label.substr(label.length - 7) == '-option'
+    )
+  }
 
-const AddressForm: StorefrontFunctionComponent<Props> = ({
-  address,
-  summary,
-}) => {
-  const { setAddress } = useAddressContext()
+  const hasOptions = (field: OptionsField) => {
+    return field && field.options
+  }
 
   const parseLineFragment = (fragment: LineFragment) => {
-    return address[fragment.name] != null ? (
+    const field = fields[fragment.name as keyof Fields]
+    const labelName = field && field.label ? field.label : null
+
+    if (ignoredFields.has(fragment.name)) return null
+    if (address[fragment.name] == null) return null
+    if (numberHasWithoutOption(labelName)) return <NumberOption showCheckbox />
```

Comment:
Wouldn't be clearer to name this function `hasWithoutNumberOption`?
```suggestion
    if (hasWithoutNumberOption(labelName)) return <NumberOption showCheckbox />
```

Reply:
Yes, it would 😅